process/heartbeat/interceptedHeartbeat.go

```diff
@@ -0,0 +1,139 @@
+package heartbeat
+
+import (
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-go-core/marshal"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/heartbeat"
+	"github.com/ElrondNetwork/elrond-go/process"
+)
+
+// ArgBaseInterceptedHeartbeat is the base argument used for messages
+type ArgBaseInterceptedHeartbeat struct {
+	DataBuff    []byte
+	Marshalizer marshal.Marshalizer
+}
+
+// ArgInterceptedHeartbeat is the argument used in the intercepted heartbeat constructor
+type ArgInterceptedHeartbeat struct {
+	ArgBaseInterceptedHeartbeat
+	PeerId core.PeerID
+}
+
+type interceptedHeartbeat struct {
+	heartbeat heartbeat.HeartbeatV2
+	payload   heartbeat.Payload
+	peerId    core.PeerID
+}
+
+// NewInterceptedHeartbeat tries to create a new intercepted heartbeat instance
+func NewInterceptedHeartbeat(arg ArgInterceptedHeartbeat) (*interceptedHeartbeat, error) {
+	err := checkBaseArg(arg.ArgBaseInterceptedHeartbeat)
+	if err != nil {
+		return nil, err
+	}
+	if len(arg.PeerId) == 0 {
+		return nil, process.ErrEmptyPeerID
+	}
+
+	hb, payload, err := createHeartbeat(arg.Marshalizer, arg.DataBuff)
+	if err != nil {
+		return nil, err
+	}
+
+	intercepted := &interceptedHeartbeat{
+		heartbeat: *hb,
+		peerId:    arg.PeerId,
+		payload:   *payload,
+	}
```

Comment:
```suggestion
	intercepted := &interceptedHeartbeat{
		heartbeat: *hb,
		payload:   *payload,
		peerId:       arg.PeerId,
	}
```
please use the same order in creation as in declaration